src/scheduler/MpiWorld.cpp

```diff
@@ -25,40 +30,109 @@ MpiWorld::MpiWorld()
   , cartProcsPerDim(2)
 {}
 
-faabric::scheduler::FunctionCallClient& MpiWorld::getFunctionCallClient(
-  const std::string& otherHost)
+void MpiWorld::sendRemoteMpiMessage(
+  int sendRank,
+  int recvRank,
+  const std::shared_ptr<faabric::MPIMessage>& msg)
 {
-    auto it = functionCallClients.find(otherHost);
-    if (it == functionCallClients.end()) {
-        // The second argument is forwarded to the client's constructor
-        auto _it = functionCallClients.try_emplace(otherHost, otherHost);
-        if (!_it.second) {
-            throw std::runtime_error("Error inserting remote endpoint");
+    // Assert the ranks are sane
+    assert(0 <= sendRank && sendRank < size);
+    assert(0 <= recvRank && recvRank < size);
```

Comment:
This assertion could fail because of an error in the application we're executing, so should probably be an error with a useful message.